modules/social_features/social_profile/social_profile.module

```diff
@@ -398,6 +391,37 @@ function social_profile_preprocess_profile(array &$variables) {
   }
 }
 
+/**
+ * Helps determine if current user has access to view the email field.
+ *
+ * @param \Drupal\profile\Entity\Profile $profile
+ *   The profile the current user is viewing.
+ * @param \Drupal\Core\Session\AccountProxyInterface $current_user
+ *   The current user.
+ *
+ * @return bool
+ *   Return TRUE if current user is allowed to see the email, FALSE otherwise.
+ */
+function social_profile_check_email_field_access(Profile $profile, AccountProxyInterface $current_user) {
+  // Prepare some variables for checking on the email field.
+  $global_show_email = \Drupal::config('social_profile.settings')->get('social_profile_show_email');
+  if (isset($profile->field_profile_show_email)) {
+    $profile_show_email = $profile->field_profile_show_email->value;
+  }
+
+  // If the current user has this permission, return true straight away.
+  if ($current_user->hasPermission('social profile privacy view hidden fields')) {
```

Comment:
How about moving this statement above all?

Reply:
Done!